golden-ami-builder/main.tf

```diff
@@ -0,0 +1,245 @@
+}
+
+resource "aws_security_group" "all_egress" {
+  count = var.security_group_ids == null ? 1 : 0
+
+  name        = "${module.golden_ami_lookup.ami_name_prefix}-image-builder"
+  description = "Security group used by Image Builder instances. Allows no inbound traffic and all outbound traffic."
+  vpc_id      = module.vpcread.vpc
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = var.tags
+}
+
+resource "aws_imagebuilder_distribution_configuration" "golden_ami" {
+  name        = "${module.golden_ami_lookup.ami_name_prefix}-distribution-configuration"
+  description = "Distribution Configuration for Amazon-Linux-2-based Golden AMI"
+
+  distribution {
+    ami_distribution_configuration {
+      ami_tags = var.tags
+      name     = "${module.golden_ami_lookup.ami_name_prefix}-{{ imagebuilder:buildDate }}"
+    }
+    region = local.region
+  }
+}
+
+resource "aws_imagebuilder_infrastructure_configuration" "golden_ami" {
+  description           = "Infrastructure configuration for Amazon-Linux-2-based golden AMI pipeline"
+  instance_profile_name = aws_iam_role.instance_profile.id
+  instance_types        = ["t3.micro"]
+  name                  = "${module.golden_ami_lookup.ami_name_prefix}-infrastructure-configuration"
+  security_group_ids = coalesce(
+    var.security_group_ids,
+    [for sg in aws_security_group.all_egress : sg.id]
+  )
+  sns_topic_arn                 = var.alerting_sns_topic_arn
+  subnet_id                     = module.vpcread.private_subnets[0]
+  terminate_instance_on_failure = true
+
+  logging {
+    s3_logs {
+      s3_bucket_name = module.image_builder_logs.bucket_id
+    }
+  }
+
+  tags = var.tags
+}
+
+resource "aws_imagebuilder_component" "download_and_install_cortex_xdr" {
+  description = "Simple component which downloads and installs Cortex XDR agent from distribution bucket"
+  data = templatefile(
+    "${path.module}/templates/download-and-install-cortex-xdr.yaml",
+    {
+      software_distribution_bucket_id = var.software_distribution_bucket_id
+    }
+  )
+  name     = "download-and-install-cortex-xdr"
+  platform = "Linux"
+  version  = "1.0.0"
+
+  tags = var.tags
+}
+
+resource "aws_imagebuilder_image_recipe" "golden_ami" {
+  description = "Recipe for Amazon-Linux-2-based Golden AMI"
+
+  block_device_mapping {
+    device_name = "/dev/xvda"
+
+    ebs {
+      kms_key_id            = data.aws_kms_key.volume_key.arn
+      delete_on_termination = false
+      volume_size           = 200
+      volume_type           = "gp2"
+    }
+  }
+
+  block_device_mapping {
+    device_name = "/dev/sdf"
+
+    ebs {
+      kms_key_id            = data.aws_kms_key.volume_key.arn
+      delete_on_termination = false
+      volume_size           = 250
+      volume_type           = "gp2"
+    }
+  }
+
+  working_directory = "/var/tmp"
```

Comment:
Maybe this is the correct way to do it, but `/var/tmp` is permanent, while `/tmp` is cleared on boot. Since we are deleting these files anyways, maybe `/tmp` would make more sense?

Reply:
yeah `/tmp` works, I think I just picked the default